tests/compile/reduction-tree.fil

```diff
@@ -10,17 +10,10 @@ comp AddRed[#W, #N, #K]<G:1>(
         #N == pow2(#K),
         #K >= 0
 {
+
     if #K == 0 {
         out = in{0};
     } else {
-        // Required to make recursive calls type check.
-        assume pow2(#K-1) == #N/2;
-        // This is a corollary of #K != 0
-        assume #N >= 2;
-        // This is not true in general for integer division but
-        // because #N is a power of 2 it is true.
-        assume #N - #N/2 == #N/2;
```

Comment:
Awesome! Maybe we should also add some tests under `tests/check` to see if assumptions are being instantiated correctly?